File: backend/app/models/edital.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import datetime
from decimal import Decimal
from uuid import UUID
from enum import Enum


class EditalStatus(str, Enum):
    publicado = "publicado"
    aberto = "aberto"
    encerrado = "encerrado"
    revogado = "revogado"
    anulado = "anulado"
    suspenso = "suspenso"


class EditalSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: Optional[UUID] = None
    pncp_id: str
    numero_compra: Optional[str] = None
    nome_orgao: Optional[str] = None
    uf: Optional[str] = None
    municipio_nome: Optional[str] = None
    modalidade_nome: Optional[str] = None
    objeto_compra: Optional[str] = None
    status: Optional[EditalStatus] = None
    valor_total_estimado: Optional[Decimal] = None
    data_publicacao_pncp: Optional[datetime] = None
    data_abertura_proposta: Optional[datetime] = None
    data_encerramento_proposta: Optional[datetime] = None
    link_edital: Optional[str] = None
    rank: Optional[float] = None


class Pagination(BaseModel):
    page: int
    page_size: int
    total_items: int
    total_pages: int


class SearchMeta(BaseModel):
    sources: List[str]
    cache_hit_count: int
    pncp_new_count: int
    search_time_ms: int


class SearchResponse(BaseModel):
    data: List[EditalSummary]
    pagination: Pagination
    meta: SearchMeta


class SearchParams(BaseModel):
    q: Optional[str] = None
    uf: Optional[str] = None
    modalidade: Optional[str] = None
    status: Optional[str] = None
    valor_min: Optional[Decimal] = None
    valor_max: Optional[Decimal] = None
    data_inicial: Optional[str] = None
    data_final: Optional[str] = None
    page: int = 1
    page_size: int = 20

File: frontend/src/types/edital.ts
export type EditalStatus = 'publicado' | 'aberto' | 'encerrado' | 'revogado' | 'anulado' | 'suspenso'

export interface EditalSummary {
  id: string | null
  pncp_id: string
  numero_compra: string | null
  nome_orgao: string | null
  uf: string | null
  municipio_nome: string | null
  modalidade_nome: string | null
  objeto_compra: string | null
  status: EditalStatus | null
  valor_total_estimado: number | null
  data_publicacao_pncp: string | null
  data_abertura_proposta: string | null
  data_encerramento_proposta: string | null
  link_edital: string | null
  rank: number | null
}

export interface Pagination {
  page: number
  page_size: number
  total_items: number
  total_pages: number
}

export interface SearchMeta {
  sources: string[]
  cache_hit_count: number
  pncp_new_count: number
  search_time_ms: number
}

export interface SearchResponse {
  data: EditalSummary[]
  pagination: Pagination
  meta: SearchMeta
}

export interface SearchFilters {
  q: string
  uf: string
  modalidade: string
  status: string
  valor_min: string
  valor_max: string
  data_inicial: string
  data_final: string
  page: number
  page_size: number
}

export interface Modalidade {
  codigo: string
  nome: string
}
